### PR TITLE
Adapting to Symfony 2.1:

### PR DESCRIPTION
**Pull request by marekkalnik**
- pass file_locator to Twig service constructor
- moved set/getLocale calls from session to request

Please, merge this in master, so we can keep it synchronized (and usable) with symfony's master.

### CODE REVIEW

File: Resources/config/services.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ services:
```
Comment (lsmith77) [inline Resources/config/services.yml:31]: FYI .. the best practice defines that reusable Bundles should use XML for their configuration.

Comment (marekkalnik) [inline Resources/config/services.yml:31]: It's already on my TODO list, thanks for the reminder though :)

**Comment by mteneur** [comment]
From a fresh symfony standard edition install (2.0.6), fatal error with the @file_locator parameter.

I had to replace the line with the old one:
    roger.templating:
        class: %templating.engine.twig.class%
        arguments: ['@roger.twig', '@templating.name_parser', '@templating.globals']

**Comment by mteneur** [comment]
Ok sorry ...didn't see pull request title 'Adapting to Symfony 2.1' .

**Comment by marekkalnik** [comment]
@mteneur Symfony 2.0 branch is available as sf2.0.x

**Pull request merged**